forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -484,4 +503,155 @@ mod test {
+            target: None,
+            node_url: None,
+            testnet: true,
+            ..Default::default()
+        };
+        let expected = cmd::Deploy {
+            target: None,
+            node_url: Some("beta-4.fuel.network/graphql".to_string()),
+            testnet: true,
+            gas: Gas { price: 1, limit: 1 },
+            ..Default::default()
+        };
+        let actual = apply_target(input).unwrap();
+        assert_cmd_eq(expected, actual);
+    }
+
+    #[test]
+    fn test_apply_target_beta4() {
+        let input = cmd::Deploy {
+            target: Some(Target::Beta4),
+            node_url: None,
+            testnet: false,
+            ..Default::default()
+        };
+        let expected = cmd::Deploy {
+            target: Some(Target::Beta4),
+            node_url: Some("beta-4.fuel.network/graphql".to_string()),
+            testnet: false,
+            gas: Gas { price: 1, limit: 1 },
+            ..Default::default()
+        };
+        let actual = apply_target(input).unwrap();
+        assert_cmd_eq(expected, actual);
+    }
+
+    #[test]
+    fn test_apply_target_url_beta4() {
+        let input = cmd::Deploy {
+            target: None,
+            node_url: Some("beta-4.fuel.network/graphql".to_string()),
+            testnet: false,
+            ..Default::default()
+        };
+        let expected = cmd::Deploy {
+            target: None,
+            node_url: Some("beta-4.fuel.network/graphql".to_string()),
+            testnet: false,
+            gas: Gas { price: 1, limit: 1 },
+            ..Default::default()
+        };
+        let actual = apply_target(input).unwrap();
+        assert_cmd_eq(expected, actual);
+    }
+
+    #[test]
+    fn test_apply_target_beta4_custom_gas() {
+        let input = cmd::Deploy {
+            target: Some(Target::Beta4),
+            node_url: None,
+            testnet: false,
+            gas: Gas {
+                price: 12,
+                limit: 13,
+            },
+            ..Default::default()
+        };
+        let expected = cmd::Deploy {
+            target: Some(Target::Beta4),
+            node_url: Some("beta-4.fuel.network/graphql".to_string()),
+            testnet: false,
+            gas: Gas {
+                price: 12,
+                limit: 13,
+            },
+            ..Default::default()
+        };
+        let actual = apply_target(input).unwrap();
+        assert_cmd_eq(expected, actual);
+    }
+
+    #[test]
+    fn test_apply_target_beta3() {
+        let input = cmd::Deploy {
+            target: Some(Target::Beta3),
+            node_url: None,
+            testnet: false,
+            ..Default::default()
+        };
+        let expected = cmd::Deploy {
+            target: Some(Target::Beta3),
+            node_url: Some("beta-3.fuel.network/graphql".to_string()),
+            testnet: false,
+            gas: Gas {
+                price: 1,
+                limit: 100000000,
```

Comment:
Added some unit tests to show the current behavior - still don't understand the logic behind overriding the gad price and limit for the testnets.

Reply:
Thanks so much for adding the tests 💯 I am not sure why this is only needed for beta4 but the default max gas limit returns:

```
Error: Decode error: Custom { kind: Other, error: "Response errors; TransactionGasLimit" }
```

This is not the case for other beta networks.

If i change the default value, beta-3 and beta-2 would break, so I patched it only for beta-4 and if users provide `gas_limit` their `gas_limit` will override our preset value (unless they specified the max value, but again providing max value for gas limit returns the error above). This is not ideal but it enables to deploy without specifying any other flags

Reply:
I'm just wondering if there's a more appropriate gas limit than 1. It seems so low. We'd want to make sure the default gas limit to work for users.

Reply:
We now ask the test net for min gas price if the user left it as 0 by default. The testnet returns the accepted min gas price and we use that instead of 0 so that the transaction is valid.

We do the same thing for gas_limit if they left as the default max value, we ask the chain the valid max value and use it instead, so that we do not exceed the limit set by the node.

I think that while deploying to test net we can override this as we want the transactions to be valid by default and if the user choses to be providing the extremes we can use the extreme set by the node instead. Maybe I can add a warning to notify them?

Reply:
On a second thought modiying this rn to make sure we do not override any user given data. If the user wants to override the default data returned by the node they can. Maybe they would want to test some stuff around that